### PR TITLE
fix: require.cache undefined

### LINKED ISSUE #2460: [Bug Report]: require.cache not available in CJS modules
### System Info

(VSCode devcontainer)
```
  System:
    OS: Linux 5.15 Debian GNU/Linux 11 (bullseye) 11 (bullseye)
    CPU: (5) arm64 unknown
    Memory: 14.74 GB / 23.45 GB
    Container: Yes
    Shell: 5.1.4 - /bin/bash
  Binaries:
    Node: 18.12.1 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 8.19.2 - /usr/local/bin/npm
  npmPackages:
    @rspack/cli: ^0.1.2 => 0.1.2 
```

### Details

`require.cache` seem not to be available in CJS modules (and is used by fastify)...

### Reproduce link

https://github.com/lalexdotcom/rspack-bug-report

### Reproduce Steps

`npm install`

Run without rspack `npm run start-js`
```
fast= function
require= require
require.cache= object
```

Run bundled by rspack `npm run start-bundled`
```
fast= function
require= __webpack_require__
require.cache= undefined
```

### PR DESCRIPTION
**Pull request by faga295**
## Summary
closed #2460 

Currently, `require.cache` will transform to `__webpack_require__.c`, but I'm confusing about where to insert `runtime_globals::MODULE_CACHE` to `runtime_requirement`.

Since `&mut runtime_requirement` is borrowed to `inject_runtime_helper`,  I can't find a good solution unless move transform `require.cache` operation to `inject_runtime_helper` and insert `MODULE_CACHE` in it.

Perhaps my confusion comes from my lack of familiarity with rust, and if so, please forgive me, I'm trying to learn.

## Types of changes

- [ ] Docs change / Dependency upgrade
- [x] Bug fix
- [ ] New feature / Improvement
- [ ] Refactoring
- [ ] Breaking change

## Checklist

- [x] I have added changeset via `pnpm run changeset`.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: crates/rspack_plugin_javascript/src/plugin.rs [hunk 1]
```diff
@@ -138,6 +138,10 @@ impl JsPlugin {
     sources.add(self.render_require(args));
     sources.add(RawSource::from("\n}\n"));
 
+    sources.add(RawSource::from(
+      "__webpack_require__.cache = __webpack_module_cache__;\n",
```
Comment (jerrykingxyz) [inline crates/rspack_plugin_javascript/src/plugin.rs:142]: directly add `cache` on __webpack_require__ is not a good idea, maybe we should replace `require.cache` to `__webpack_require__.c` as same as webpack

Comment (faga295) [reply crates/rspack_plugin_javascript/src/plugin.rs:142]: Yes, this issue obviously shouldn’t be resolved like this. I will find a solution as soon as possible.

**Comment by underfin** [comment]
@faga295 Could you refactor the code using const dependency? here is an example https://github.com/web-infra-dev/rspack/blob/main/crates/rspack_plugin_javascript/src/visitors/dependency/scanner.rs#L336

File: crates/rspack_plugin_javascript/src/visitors/dependency/scanner.rs [hunk 2]
```diff
@@ -397,6 +397,31 @@ impl VisitAstPath for DependencyScanner<'_> {
...
+  fn visit_member_expr<'ast: 'r, 'r>(
+    &mut self,
+    n: &'ast MemberExpr,
+    ast_path: &mut AstNodePath<AstParentNodeRef<'r>>,
+  ) {
+    if let Expr::Ident(obj_ident) = &*n.obj {
+      if "require".eq(&obj_ident.sym) {
+        if let MemberProp::Ident(prop_ident) = &n.prop {
+          if "cache".eq(&prop_ident.sym) {
+            println!("ident: {:?}", prop_ident);
```
Comment (jerrykingxyz) [inline crates/rspack_plugin_javascript/src/visitors/dependency/scanner.rs:410]: should remove debug code

File: crates/rspack_core/src/dependency/const_dependency.rs [hunk 3]
```diff
@@ -34,11 +37,31 @@ impl CodeGeneratable for ConstDependency {
...
 
     let expr = self.expression.clone();
-    cgr.visitors.push(
-      create_javascript_visitor!(exact &self.ast_path, visit_mut_expr(n: &mut Expr) {
-        *n = expr.clone();
-      }),
-    );
+
+    let mut is_member = false;
+    if let Some(path) = self.ast_path.last() {
```
Comment (jerrykingxyz) [inline crates/rspack_core/src/dependency/const_dependency.rs:42]: This code seems hard to read, here are some suggested solutions
* check member expr in `scanner -> visit_expr` and const_dependency no need to change
* or add a type field in const_dependency

cc @underfin

Comment (faga295) [reply crates/rspack_core/src/dependency/const_dependency.rs:42]: > check member expr in scanner -> visit_expr and const_dependency no need to change

I think this is better.

**Review by underfin — DISMISSED** [review]

**Comment by underfin** [comment]
> faga seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).
@faga295 Look like cla is not signed. Look like your name of commit is not equal to your github name, Could you check it?

**Comment by underfin** [comment]
@faga295 Look like you need to rebase the main branch, because `runtime_globals` moved to `RuntimeGlobals`. Look like cargo clippy crashed in ci, you can run `cargo clean && cargo clippy` in local environment to check it.

**Comment by underfin** [comment]
@faga295 I fixed clippy error at my commit. It colud run ci success.

**Comment by underfin** [comment]
Thanks a lot!

**Review by underfin — APPROVED** [review]

**Pull request merged**